rdtools/test/filtering_test.py

```diff
@@ -1,96 +1,298 @@
+    time_range_3 = pd.date_range(power_datetime_index.index.max(),
+                                 '2017-06-06T07:00:00.000Z', freq='5T')
+    power_datetime_index_3.index = pd.to_datetime(time_range_3[:100])
+    power_datetime_index_3 = power_datetime_index_3.iloc[1:]
+    power_datetime_index = pd.concat([power_datetime_index,
+                                      power_datetime_index_3])
+    # Note: Power is expected to be Series object with a datetime index.
+    return power_datetime_index
 
-    def setUp(self):
-        self.tcell = np.array([-50, -49, 0, 109, 110])
 
-    def test_tcell_filter(self):
-        filtered = tcell_filter(self.tcell,
-                                temperature_cell_low=-50,
-                                temperature_cell_high=110)
+@pytest.fixture
+def generate_power_time_series_one_min_intervals():
+    power_datetime_index = pd.Series(np.arange(1, 51))
+    power_datetime_index = pd.concat([power_datetime_index,
+                                      power_datetime_index[::-1]])
+    # Add datetime index to second series
+    time_range = pd.date_range('2016-12-02T11:00:00.000Z',
+                               '2017-06-06T07:00:00.000Z', freq='1T')
+    power_datetime_index.index = pd.to_datetime(time_range[:100])
+    # Note: Power is expected to be Series object with a datetime index.
+    return power_datetime_index
 
-        # Expected high and low tcell cutoffs to be non-inclusive.
-        expected_result = np.array([False, True, True, True, False])
-        self.assertListEqual(filtered.tolist(), expected_result.tolist())
 
+@pytest.fixture
+def generate_power_time_series_clipping():
+    power_no_datetime_index = pd.Series(np.arange(1, 51))
+    power_no_datetime_index = pd.concat([power_no_datetime_index,
+                                         power_no_datetime_index[::-1]])
+    power_no_datetime_index[48:52] = 50
+    power_no_datetime_index = power_no_datetime_index.reset_index(drop=True)
+    power_datetime_index = power_no_datetime_index.copy()
+    # Add datetime index to second series
+    time_range = pd.date_range('2016-12-02T11:00:00.000Z',
+                               '2017-06-06T07:00:00.000Z', freq='H')
+    power_datetime_index.index = pd.to_datetime(time_range[:100])
+    # Note: Power is expected to be Series object with a datetime index.
+    return power_no_datetime_index, power_datetime_index
 
-class ClipFilterTestCase(unittest.TestCase):
+
+def test_quantile_clip_filter():
     ''' Unit tests for inverter clipping filter.'''
+    power = pd.Series(np.arange(1, 101))
+    # Note: Power is expected to be Series object because clip_filter makes
+    #       use of the Series.quantile() method.
+    filtered = quantile_clip_filter(power, quantile=0.98)
+    # Expect 99% of the 98th quantile to be filtered
+    expected_result = power < (98 * 0.99)
+    assert ((expected_result == filtered).all())
+
+
+def test_logic_clip_filter(generate_power_time_series_no_clipping,
```

Comment:
These three test functions are pretty similar except for which function is being run, would it make sense to have a single test function [parametrized](https://docs.pytest.org/en/6.2.x/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions) across the filter functions to be more [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)?